src/NosnocSolver.m

```diff
@@ -133,26 +146,15 @@ function set(obj, type, val)
             violation = max([lbg_violation, ubg_violation, lbw_violation, ubw_violation]);
         end
 
-        function converged = is_converged(obj, stats)
-            if strcmp(obj.solver_options.solver, 'ipopt')
-                last_stats = stats.solver_stats(end);
-                converged = 0;
-                if isfield(last_stats, 'iterations') && ~isempty(last_stats.iterations)
-                    inf_pr = last_stats.iterations.inf_pr(end);
-                    inf_du = last_stats.iterations.inf_du(end);
-                    if stats.complementarity_stats(end) < 10 * obj.solver_options.comp_tol
-                        converged = 1;
-                    end
-                else
-                    converged = 0;
-                end
-            else
-                % TODO..
-                converged = [];
+        function converged = comp_tol_met(obj, stats)
```

Comment:
I think renaming this function makes it less readable (`is_converged` -> `comp_tol_met`).
More verbose function names would be good IMO.

Reply:
This function is no longer the only thing being checked to check for convergence hence the rename to more accurately reflect exactly what it checks.

Reply:
I see now. Maybe rename the flag to just `flag` and the function to `complementarity_tol_met`.
I thought something like compute tolerance metric :sweat_smile:

Reply:
:100: I see.... that actually is confusing, good catch :sweat_smile: